### DISCUSSION TITLE
INSTALLING OLDER LABS FIRMWARE?

### DISCUSSION THREAD

**Discussion by IRCGraphic** [gopro/labs#527] [Q&A]
Hello, I have a GoPro 11 with the latest Labs and i don't know if it's causing problems corrupting stabilisation data that later i use to stabilise videos in the PC, I have previous experiences with GoPro's main firmwares sometimes introducing bugs in certain features and as a matter of fact also with malfunctioning units that need replacement, so in the best of the cases I would like to know if it's a FW related issue creating problems to my unit and be able to install a stable FW.
So i would like to install the previous GoPro 11 Labs FW but I see it's not available, instead I tried to install a GoPro 11 Labs from december 2022 but the device says something like "failed installation" or something similar.
I would appreciate any help and advice as for now I can't record video without overcoming this issue, thanks.

**Comment by dnewman-gpsw** [comment]
Always report details on Labs interfering with stock camera modes, as it should not.  Labs is so intended to be only additions.

Can you share examples of the stabilization issue?

Non-Labs FW is available here https://gopro.com/en/us/update/hero11-black

Unfortunately you can't downgrade to older than 2.12 firmware.  Labs or not. 

They are upgraded Labs builds coming, so details on the bug would helpful get that fix included.

**Reply by IRCGraphic** [reply]
I will be glad to collaborate, i think first i should explain the context don't be at the end that it's rather a malfunctioning unit that needs replacement instead of the FW; 
So I'm using an app called Gyroflow to stabilise a GoPro 11 video with ERS disabled and using the Max fisheye in 4K 25fps 4:3, Gyroflow is a robust and reliable app I always use to stabilise many GoPro videos without problem as it reads automatically GoPro's stabilisation data in the video, but for this GoPro 11 + !MBERS=1 + Max Lens setup I use a special method specified by Gyroflow software support that if need be I can detail and provide the lens profile and video samples to reproduce the problem. This issue after contacting Gyroflow support it's in their words: "I've seen some corrupted data like that, and one user reported that going back to stock firmware (not labs) fixed the issue, so it's likely a firmware fault, I also already sent a few sample files to GoPro, they are aware of this bug, but it's rare and looks like they still weren't able to fix it"
As this GoPro setup workflow needs to use raw gyro data for correcting for rolling shutter as the default synchronised quaternion that GoPro uses is per-frame, which is not enough for rolling shutter correction, so then the raw gyro data needs to be synched with the video to stabilise it, the problem arises when the sync values are not consistent which should be in any normal GoPro video, Gyroflow graphic of the sync point issue here (the sync point values should be consistent in a normal video instead of varying so much along the video): 
![image](https://github.com/gopro/labs/assets/113941057/b34d82b9-4211-4cb8-a5ab-5f6b4223aaf6)

**Reply by dnewman-gpsw** [reply]
I'm not aware of this bug, but I isn't likely Labs (maybe a 10% probability.) It would be hard for Labs to mess with Gyro data and video timing. It would be interesting see what the corrupt gyro data looks like.

**Reply by IRCGraphic** [reply]
Yeah I forgot to mention that later Gyroflow also commented something that actually normal FW does that too.
So here is a problematic sample for that Gyroflow workflow along its lens profile for the app to correct a Max lens in 4:3 (in the default wide lens video setting (without activating the Max lens mode): https://drive.google.com/drive/folders/1yfOEXc933gZXIFHVOZzzEZc2GdF3GtcO?usp=sharing

-Malfunctions context: this video's GoPro 11 unit very early at the first long video recordings started presenting shutdowns while recording barely a few minutes or so (full battery and full 128 gb Sandisk sd card) for now (I'm completely saturated to ask for another replacement) I can bypass the issue by activating sound, so i know if stops recording, then it doesn't shutdown anymore but tends to get unresponsive behaviour when finally I stop recording and once i even had to force a reboot (both buttons pressed) and started then "repairing the file". In this video all went smooth and it presented none of these issues while the recording, also besides the shutdown and unresponsiveness issues all the videos looking normal without any apparent problems.

-Gyroflow instructions: if you want to test the sample in Gyroflow, load the video, then load the provided lens profile (at this point you could play a good stabilised video only problem the rolling shutter it's not fully corrected), then select integration method as VQF, select Auto sync with the default values and afterwards it should do a good stabilisation but it totally fails, also for reference the FOV value is meant to be around 2 or less, then "dynamic zoom" should be deactivated as "no zoom" and "output size" as 16:9.

I'd really appreciate your expert opinion, thanks.

**Reply by dnewman-gpsw** [reply]
I'm not sure what failure you are seeing in Gyroflow, I'm not super experienced in the tool, but the data within the file looks good, and the video seems to stabilize.

**Reply by IRCGraphic** [reply]
Well Gyroflow checked it and immediately recognised it as a GoPro bug that apparently it does not occur in all cameras.
The code MACR= could potentially bypass all this as the ERS could be active in-camera and there would be no need for using the raw gyro data, the question is what barrel correction model is using? Coefficients could be calculated with a tool using the same model then just writing into the MACR= code

**Reply by dnewman-gpsw** [reply]
Where is that error reported? The clip you sent worked here in gyroflow (latest version.)

MACR will not likely work, as it was for a more subtle prescription change.

**Reply by IRCGraphic** [reply]
A pity about MACR it's awesome to unwrap internally.
The bug as said by Gyroflow support it's not Labs but normal FW too, i and they forgot to mention it at the beginning as they where vague in details until i asked for it. They say it's known and reported to GoPro, i guess developers must use a way of sharing that info.

Believe me i tried 3 versions of Gyroflow before bothering their support and finally they also reproduced the stabilisation fail, the instructions i detailed for setting up the stabilisation are important to reproduce it: loading lens profile, changing integration method to VQF for a rolling shutter corrected stabilisation, hit the Auto sync with default values and just the first seconds of the video are a complete mess already. I can upload a video of it if needed for reference.

**Reply by dnewman-gpsw** [reply]
Other than some wobbles at the very beginning, the stabilization seems to work, the GYRO frequency is solid, and no frames where dropped, so GYRO and Motion Vectors should be syncable.  Seems not an issue, so I might need to see your render to understand, although there is very little Labs team can do if it is in the core camera.

**Reply by IRCGraphic** [reply]
Sorry for the Labs FW confusion I barely have information to understand what's going on, only asking for insights, which I'm grateful GoPro Labs welcomes this advanced aspects of video. Said by the way i find the MACR code is an interesting idea for a possible Labs feature request to input lens profiling, I mean an internal lens calibration surely has it's benefits like not needing to transcode in post to stabilise just use the original compression with an anamorphic or a fisheye or whatever improved optics. Thanks
